terraform/modules/SplitGrantsGovXMLDB/main.tf

```diff
@@ -41,23 +41,24 @@ module "lambda_execution_policy" {
         "${data.aws_s3_bucket.source_data.arn}/sources/*/*/*/grants.gov/extract.xml"
       ]
     }
-    AllowInspectS3PreparedData = {
+    AllowReadDynamoDBPreparedData = {
       effect = "Allow"
       actions = [
-        "s3:GetObject",
-        "s3:ListBucket"
-      ]
-      resources = [
-        data.aws_s3_bucket.prepared_data.arn,
-        "${data.aws_s3_bucket.prepared_data.arn}/*/*/grants.gov/v2.xml"
+        "dynamodb:GetItem",
+        "dynamodb:ListTables",
       ]
+      resources = [var.grants_prepared_dynamodb_table_arn]
     }
     AllowS3UploadPreparedData = {
       effect  = "Allow"
       actions = ["s3:PutObject"]
       resources = [
-        # Path: <first 3 digits of grant ID>/<grant id>/grants.gov/v2.xml
-        "${data.aws_s3_bucket.prepared_data.arn}/*/*/grants.gov/v2.xml"
+        # Path: <first 3 digits of grant ID>/<grant id>/grants.gov/v2.xml (deprecated)
+        "${data.aws_s3_bucket.prepared_data.arn}/*/*/grants.gov/v2.xml",
+        # Path: <first 3 digits of grant ID><grant id>/grants.gov/v2.OpportunitySynopsisDetail_1_0.xml
+        "${data.aws_s3_bucket.prepared_data.arn}/*/*/grants.gov/v2.OpportunitySynopsisDetail_1_0.xml",
+        # Path: <first 3 digits of grant ID><grant id>/grants.gov/v2.OpportunityForecastDetail_1_0.xml
+        "${data.aws_s3_bucket.prepared_data.arn}/*/*/grants.gov/v2.OpportunityForecastDetail_1_0.xml",
```

Comment:
This fixes the permissions issue the `SplitGrantsGovXMLDB` is currently experiencing in Staging.